server/src/domain/person/person.service.ts

```diff
@@ -62,17 +67,19 @@ export class PersonService {
   }
 
   async getAssets(authUser: AuthUserDto, id: string): Promise<AssetResponseDto[]> {
-    const assets = await this.repository.getAssets(authUser.id, id);
+    await this.access.requirePermission(authUser, Permission.PERSON_READ, id);
+    const assets = await this.repository.getAssets(id);
     return assets.map(mapAsset);
   }
 
   async update(authUser: AuthUserDto, id: string, dto: PersonUpdateDto): Promise<PersonResponseDto> {
     let person = await this.findOrFail(authUser, id);
 
     if (dto.name !== undefined || dto.birthDate !== undefined || dto.isHidden !== undefined) {
+      await this.access.requirePermission(authUser, Permission.PERSON_WRITE, id);
```

Comment:
Doing anything in this method should require PERSON_WRITE, including the thumbnail change. Can you move this to the top, before the find or fail?